### ISSUE TITLE
Cannot use 3dForceGraph via NPM and import

### ISSUE THREAD

**Issue by fedevela** [vasturiano/3d-force-graph#10]
Hello Mr. Vasturiano. Thank you very much for this wonderful framework! I am working on learning it an hope to be able to contribute back soon!!! Thank you again for all you do.

I have been able to use 3d-force-graph like this:
```
    <script src="3d-force-graph.js"></script>
```

But now I am trying to set up 3d-force-graph in a React app with NPM, i am doing thus:

```
import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import ForceGraph3D from '3d-force-graph';

class App extends Component {
  render() {

    const Graph = ForceGraph3D()
    (document.getElementById("3d-graph"));  

    const N = 3000;
    const gData = {
        nodes: [...Array(N).keys()].map(i => ({ id: i })),
        links: [...Array(N).keys()]
            .filter(id => id)
            .map(id => ({
                source: id,
                target: Math.round(Math.random() * (id-1))
            }))
    };

    Graph
        .cooldownTicks(300)
        .cooldownTime(20000)
        .autoColorBy('group')
        .forceEngine('ngraph')
        .graphData(gData);


    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <p className="App-intro">
          To get started, edit <code>src/App.js</code> and save to reload.
        </p>
        <div id="3d-graph"></div>
      </div>
    );
  }
}
export default App;
```
Which is according to the documentation AFAIK.

However the error I am getting is this:
```
TypeError: __WEBPACK_IMPORTED_MODULE_3_3d_force_graph___default(...) is not a function
render
src/App.js:9

   6 | class App extends Component {
   7 |   render() {
   8 | 
>  9 |     const Graph = ForceGraph3D()
  10 |     (document.getElementById("3d-graph"));  
  11 | 
  12 |     const N = 3000;
```

Any pointers or suggestions would be highly appreciated!

Thank you in advance!

**Comment by mhebrard** [comment]
same trouble with npm import...

**Comment by fedevela** [comment]
In my case it might be related to create react app. Investigating.

**Comment by shackbarth** [comment]
It worked for me right out of the box. I'm not using create-react-app, and I'm running this component through babel and webpack. I'd recommend you try replacing
```
import ForceGraph3D from '3d-force-graph';
```
with 
```
const ForceGraph3D = require("3d-force-graph");
```
to troubleshoot if it's just a problem with the way that your webpack is handling the ES6 module transpilation.

**Comment by fedevela** [comment]
@shackbarth what version of npm are you using?

**Comment by fedevela** [comment]
I was not able to load using require nor import. I get the same error in both, something along the lines of "ForceGraph3D() is not a function". So this issue is still open.

However what I did was use a normal html script tag in which I loaded the javascript file like such:
```
    <script src="3d-force-graph.js"></script>
```

Then right below it i execute this:
```
window.ForceGraph3D = ForceGraph3D;
```
And in my react component
```
          <div id="3d-graph" ref="GraphDIV" className="App-3d-graph"></div>
```

That way i can access it from my react component:

```
  componentDidMount(){
    const Graph = window.ForceGraph3D()
    (this.refs.GraphDIV);
```

Hope that workaround helps somebody!

Cheers!

**Comment by mhebrard** [comment]
didn't succeed to use the workaround...

in my component
```
componentWillMount() {
    // Include 3d-force-graph by script tag
    const script = document.createElement('script');
    script.src = 'https://unpkg.com/3d-force-graph';
    script.async = false;
    document.body.appendChild(script);
}
```
(script.async = true do not change the behavior)
Then
```
componentDidMount() {
    const myGraph = window.ForceGraph3D()(document.getElementById('scene'));
}
```

I still have the error "ForceGraph3D is not a function"

strange thing is in the web console, i get: 
```
>>window.ForceGraph3D
function kl</</</</e.default/<()
>>window.ForceGraph3D()
function t()
```

**Comment by vasturiano** [comment]
Thanks for reporting this issue. The main reason behind this error was that the file extension for the ES modules bundle (used with import) was .mjs, and the webpack config that ships with create-react-app was misinterpreting the file type due to this extension. In the just released version `1.10.2` the file extension is renamed to .module.js, so if you upgrade you shouldn't experience the issue anymore. 

Also, if you wish to integrate 3d-force-graph with React may I suggest an alternative (simpler?) way using [react-kapsule](https://github.com/vasturiano/react-kapsule). The code for something similar to your example would look like:
```
import React from 'react';
import ForceGraph3d from '3d-force-graph';
import fromKapsule from 'react-kapsule';

const Graph = fromKapsule(ForceGraph3d);

export default () => {
  const gData = { ... };

  return <Graph 
    graphData={gData}
    cooldownTicks={300}
    cooldownTime={20000}
    autoColorBy="group"
    forceEngine="ngraph"
  />;
};
```

Let me know if this fixes it for you and thanks for your patience in waiting on this issue!

**Comment by fedevela** [comment]
Thank you Mr. @vasturiano will evaluate and let you know. For now I found a way to make it work with jquery but REACT is at the top of my list!!!

Thank you very much!!!!
